test/TestProjects/MgmtDiscriminator/Generated/MgmtDiscriminatorModelFactory.cs

```diff
@@ -0,0 +1,317 @@
+        /// Please note <see cref="DeliveryRuleAction"/> is the base class. According to the scenario, a derived class of the base class might need to be assigned here, or this property needs to be casted to one of the possible derived classes.
+        /// The available derived classes include <see cref="DeliveryRuleCacheExpirationAction"/>, <see cref="DeliveryRuleCacheKeyQueryStringAction"/>, <see cref="DeliveryRuleRequestHeaderAction"/>, <see cref="DeliveryRuleResponseHeaderAction"/>, <see cref="OriginGroupOverrideAction"/>, <see cref="DeliveryRuleRouteConfigurationOverrideAction"/>, <see cref="UrlRedirectAction"/>, <see cref="UrlRewriteAction"/> and <see cref="UrlSigningAction"/>.
+        /// </param>
+        /// <returns> A new <see cref="Models.DeliveryRuleProperties"/> instance for mocking. </returns>
+        public static DeliveryRuleProperties DeliveryRuleProperties(int? order = null, DeliveryRuleCondition conditions = null, IEnumerable<DeliveryRuleAction> actions = null, IDictionary<string, DeliveryRuleAction> extraMappingInfo = null)
+        {
+            actions ??= new List<DeliveryRuleAction>();
+            extraMappingInfo ??= new Dictionary<string, DeliveryRuleAction>();
+
+            return new DeliveryRuleProperties(order, conditions, actions?.ToList(), extraMappingInfo);
+        }
+
+        /// <summary> Initializes a new instance of DeliveryRuleCondition. </summary>
+        /// <param name="name"> The name of the condition for the delivery rule. </param>
+        /// <param name="foo"> For test. </param>
+        /// <returns> A new <see cref="Models.DeliveryRuleCondition"/> instance for mocking. </returns>
+        public static DeliveryRuleCondition DeliveryRuleCondition(string name = null, string foo = null)
+        {
+            return new UnknownDeliveryRuleCondition(name, foo);
+        }
+
+        /// <summary> Initializes a new instance of DeliveryRuleAction. </summary>
+        /// <param name="name"> The name of the action for the delivery rule. </param>
+        /// <param name="foo"> for test. </param>
+        /// <returns> A new <see cref="Models.DeliveryRuleAction"/> instance for mocking. </returns>
+        public static DeliveryRuleAction DeliveryRuleAction(string name = null, string foo = null)
+        {
+            return new DeliveryRuleAction(name, foo);
+        }
+
+        /// <summary> Initializes a new instance of DeliveryRuleRemoteAddressCondition. </summary>
+        /// <param name="foo"> For test. </param>
+        /// <param name="parameters"> Defines the parameters for the condition. </param>
+        /// <returns> A new <see cref="Models.DeliveryRuleRemoteAddressCondition"/> instance for mocking. </returns>
+        public static DeliveryRuleRemoteAddressCondition DeliveryRuleRemoteAddressCondition(string foo = null, RemoteAddressMatchConditionParameters parameters = null)
+        {
+            return new DeliveryRuleRemoteAddressCondition("RemoteAddress", foo, parameters);
```

Comment:
any reason we don't just reference the enum value directly here?

Reply:
Well no particular reason just because it is simple.
I will have some investigation to see if we could use the enum instance here instead of the raw value.

Reply:
This is fixed now